app/src/main/java/ch/epfl/sweng/SDP/game/RankingFragment.java

```diff
@@ -51,21 +73,37 @@ public View onCreateView(LayoutInflater inflater,
 
         // Retrieve the ranking array, passed as argument on instantiation of the class
         roomID = getArguments().getString("roomID");
+        this.drawings =(Bitmap[]) getArguments().getParcelableArray("drawings");
+        this.playerNames = getArguments().getStringArray("playerNames");
         return inflater.inflate(R.layout.ranking_list_fragment, container, false);
     }
 
     @Override
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
+
         rankingRef = Database.getReference(TOP_ROOM_NODE_ID + "." + roomID + ".ranking");
         finishedRef = Database.getReference(TOP_ROOM_NODE_ID + "." + roomID + ".finished");
+        Typeface typeMuro = Typeface.createFromAsset(getActivity().getAssets(), "fonts/Muro.otf");
+        Button button = getActivity().findViewById(R.id.button);
+        button.setTypeface(typeMuro);
+
         retrieveFinalRanking();
     }
 
     public List<String> getRankedUsernames() {
         return rankedUsernames;
     }
 
+    private int getIndexForUserName(String username) throws IndexOutOfBoundsException {
```

Comment:
You don't need to add throws here, as in the RankingAdapter you don't really need to catch the exception.

Reply:
Yes, what else do you think I should do?

Reply:
If you remove the throws (since you already removed the try-catch) and you make the small modification I suggested inside updateUserStats you’re good to go!